src/main/scala/nodes/learning/LBFGS.scala

```diff
@@ -130,22 +130,27 @@ object LBFGSwithL2 extends Logging {
  * @param regParam L2 regularization
  * @param numIterations max number of iterations to run
  */
-class DenseLBFGSwithL2(
+class DenseLBFGSwithL2[T <: Vector[Double]](
     val gradient: Gradient.DenseGradient,
     val fitIntercept: Boolean = true,
     val numCorrections: Int = 10,
     val convergenceTol: Double = 1e-4,
     val numIterations: Int = 100,
     val regParam: Double = 0.0)
-  extends LabelEstimator[DenseVector[Double], DenseVector[Double], DenseVector[Double]] {
+  extends LabelEstimator[T, DenseVector[Double], DenseVector[Double]] {
+
+  def fit(data: RDD[T], labels: RDD[DenseVector[Double]]): LinearMapper[T] = {
+    val denseData = data.map {
+      case denseVec: DenseVector[Double] => denseVec
+      case vec => vec.toDenseVector
+    }
 
-  def fit(data: RDD[DenseVector[Double]], labels: RDD[DenseVector[Double]]): LinearMapper = {
     if (fitIntercept) {
-      val featureScaler = new StandardScaler(normalizeStdDev = false).fit(data)
+      val featureScaler = new StandardScaler(normalizeStdDev = false).fit(denseData)
```

Comment:
So, some background on why I did this sparse and dense unification stuff:
Because an optimizable estimator can only output a single estimator (as opposed to a transformer chained to an estimator), the optimizable linear solver I wrote can only handle both sparse and dense if all of the estimator choices can handle both sparse and dense. This means both `DenseLBFGSWithL2` and `SparseLBFGSWithL2` need to be able to take either sparse or dense data as input.